plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -358,8 +359,11 @@ public DeltaLakeTableHandle getTableHandle(ConnectorSession session, SchemaTable
 
         TableSnapshot tableSnapshot = metastore.getSnapshot(tableName, session);
         Optional<MetadataEntry> metadata = metastore.getMetadata(tableSnapshot, session);
-        if (metadata.isPresent() && getColumnMappingMode(metadata.get()) != DeltaLakeSchemaSupport.ColumnMappingMode.NONE) {
-            throw new TrinoException(NOT_SUPPORTED, format("Only 'none' is supported for '%s' table property", COLUMN_MAPPING_MODE_CONFIGURATION_KEY));
+        if (metadata.isPresent()) {
+            ColumnMappingMode columnMappingMode = getColumnMappingMode(metadata.get());
+            if (columnMappingMode != ColumnMappingMode.NAME && columnMappingMode != ColumnMappingMode.NONE) {
```

Comment:
Wouldn't it be simpler to check if mode is UNKNOWN ? Or maybe I am missing something

Reply:
There will also be `id` at some point, which is not supported. Doing it this way defaults new values to failing explicitly, which is what we want.

Reply:
Ok maybe I don't get it. As far as I understand `id` would get translated to `UNKNOWN` and would fail too. The only difference would be in the exception. And this is what we care about ?